test/language/statements/switch/syntax/redeclaration/async-function-declaration-attempt-to-redeclare-with-function-declaration.js

```diff
@@ -0,0 +1,20 @@
+// This file was procedurally generated from the following sources:
+// - src/declarations/redeclare-with-function-declaration.case
+// - src/declarations/redeclare-allow-sloppy-function/switch-attempt-to-redeclare-async-function-declaration.template
+/*---
+description: redeclaration with FunctionDeclaration (AsyncFunctionDeclaration in SwitchStatement)
+esid: sec-switch-statement-static-semantics-early-errors
+flags: [generated, async-functions]
+negative:
+  phase: early
+  type: SyntaxError
+info: |
+    SwitchStatement : switch ( Expression ) CaseBlock
+
+    It is a Syntax Error if the LexicallyDeclaredNames of CaseBlock contains any
+    duplicate entries.
+
+---*/
+
+
+switch (0) { case 1: async function f() {} default: function f() {} }
```

Comment:
@anba, I'm having a hard time to find the spec parts saying the function (and async function) declaration names are added to the Lexical Scope. 

Can you help me pointing to where I'm missing there? Thanks.